### PR TITLE
Use `Path.toUri` for `hive.metastore.catalog.dir` in DeltaLakeQueryRunner 

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Use `Path.toUri` for `hive.metastore.catalog.dir` in DeltaLakeQueryRunner 

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

**Comment by ebyhr** [comment]
CI hit #13507

**Comment by findepi** [comment]
Does this allow creating tables & schemas when running `DeltaLakeQueryRunner.main` ?

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/DeltaLakeQueryRunner.java [hunk 1]
```diff
@@ -106,7 +106,7 @@ public DistributedQueryRunner build()
                 if (!deltaProperties.containsKey("hive.metastore.uri")) {
                     Path dataDir = queryRunner.getCoordinator().getBaseDataDir().resolve(DELTA_CATALOG);
                     deltaProperties.put("hive.metastore", "file");
-                    deltaProperties.put("hive.metastore.catalog.dir", dataDir.toString());
+                    deltaProperties.put("hive.metastore.catalog.dir", "file://" + dataDir);
```
Comment (findepi) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/DeltaLakeQueryRunner.java:109]: this is not the correct conversion from path (which eg can contain spaces) to a URI (which eg mustn't contain spaces, using percent-encoding instrad)

Comment (ebyhr) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/DeltaLakeQueryRunner.java]: Changed to `toUri()` instead.

**Comment by ebyhr** [comment]
> Does this allow creating tables & schemas when running DeltaLakeQueryRunner.main ?

Yes.

**Review by findepi — APPROVED** [review]

**Pull request merged**